src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,225 @@
+            insertDriversToDataBase(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT cars.id AS car_id, model, "
+                + "name, country, manufacturers.id AS manufacturer_id "
+                + "FROM cars "
+                + "JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.id = ? AND cars.is_deleted = FALSE";
+        Car car = new Car();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarByResultSet(resultSet);
+            }
+            car.setDrivers(getDriversByCarId(id));
+            return Optional.of(car);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get car by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT car.id AS car_id, model, "
+                + "name, country, manufacturer.id AS manufacturer_id "
+                + "FROM cars car "
+                + "JOIN manufacturers manufacturer "
+                + "ON c.manufacturer_id = manufacturer.id "
+                + "WHERE car.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarByResultSet(resultSet);
+                car.setDrivers(getDriversByCarId(car.getId()));
+                cars.add(car);
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get all cars from db.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = 0;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(3, car.getId());
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            updateDriversForCar(car);
+            return car;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update car in database " + car, e);
+        }
```

Comment:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
                 PreparedStatement statement = connection.prepareStatement(query)) {
            statement.setLong(3, car.getId());
            statement.setString(1, car.getModel());
            statement.setLong(2, car.getManufacturer().getId());
            statement.executeUpdate();
        } catch (SQLException e) {
            throw new RuntimeException("Can't update car in database " + car, e);
        }
        updateDriversForCar(car);
        return car;
```
You should not open a Connection inside another Connection